java/cdk/src/main/java/sleeper/cdk/stack/bulkexport/BulkExportStack.java

```diff
@@ -0,0 +1,243 @@
+        setQueueOutputProps(instanceProperties, bulkExportProcessorQ, bulkExportProcessorDlq, QueueType.PROCESSOR);
+
+        /*
+         * Output the role of the lambda as a property so that clients that want the
+         * results of queries written
+         * to their own SQS queue can give the role permission to write to their queue
+         */
+        IRole role = Objects.requireNonNull(lambda.getRole());
+        instanceProperties.set(CdkDefinedInstanceProperty.BULK_EXPORT_PROCESSOR_LAMBDA_ROLE, role.getRoleName());
+
+        CfnOutputProps bulkExportSplitterLambdaRoleOutputProps = new CfnOutputProps.Builder()
+                .value(lambda.getRole().getRoleArn())
+                .exportName(instanceProperties.get(ID) + "-" + BULK_EXPORT_PROCESSOR_LAMBDA_ROLE_ARN)
+                .build();
+        new CfnOutput(this, BULK_EXPORT_PROCESSOR_LAMBDA_ROLE_ARN, bulkExportSplitterLambdaRoleOutputProps);
+    }
+
+    /**
+     * Create a queue and a dead letter queue for the queue.
+     *
+     * @param id                 the id of the queue
+     * @param instanceProperties the instance properties
+     * @return the queue and the dead letter queue
+     */
+    private List<Queue> createQueueAndDeadLetterQueue(String id, InstanceProperties instanceProperties) {
+        String instanceId = Utils.cleanInstanceId(instanceProperties);
+        String queueNameDLQ = String.join("-", "sleeper", instanceId, id, "DLQ");
+        Queue queueDLQ = Queue.Builder
+                .create(this, id + "DeadLetterQueue")
+                .queueName(queueNameDLQ)
+                .build();
+        DeadLetterQueue deadLetterQueue = DeadLetterQueue.builder()
+                .maxReceiveCount(1)
+                .queue(queueDLQ)
+                .build();
+
+        String queueName = String.join("-", "sleeper", instanceId, id, "Q");
+        Queue queue = Queue.Builder
+                .create(this, id + "Queue")
+                .queueName(queueName)
+                .deadLetterQueue(deadLetterQueue)
+                .visibilityTimeout(
+                        Duration.seconds(instanceProperties.getInt(
+                                BULK_EXPORT_PROCESSOR_QUEUE_VISIBILITY_TIMEOUT_IN_SECONDS)))
+                .build();
+        return Arrays.asList(queue, queueDLQ);
+    }
+
+    private void attachPolicy(IFunction lambda, String id) {
+        PolicyStatementProps policyStatementProps = PolicyStatementProps.builder()
+                .effect(Effect.ALLOW)
+                .actions(
+                        Arrays.asList(SQSActions.SendMessage.getActionName(),
+                                SQSActions.ReceiveMessage.getActionName(),
+                                S3Actions.PutObject.getActionName(),
+                                S3Actions.GetObject.getActionName(),
+                                DynamoDBv2Actions.Query.getActionName()))
+                .resources(Collections.singletonList("*"))
```

Comment:
This is giving permissions for all SQS queues, S3 buckets and DynamoDB tables. Please can we grant for the specific queues, buckets and tables this needs access to? Do you want to raise this as a separate issue?